src/internal_modules/roc_address/print_supported.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2019 Roc Streaming authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include "roc_address/print_supported.h"
+#include "roc_address/protocol_map.h"
+#include "roc_core/log.h"
+#include "roc_core/printer.h"
+
+namespace roc {
+namespace address {
+
+namespace {
+
+enum { ArraySize = 100, LineSize = 70 };
+
+void print_string_list(core::Printer& prn,
+                       Interface interface,
+                       const core::StringList& list,
+                       const char* prefix,
+                       const char* suffix) {
+    const char* str = list.front();
+
+    while (str != NULL) {
+        prn.writef(" ");
+
+        size_t size = 0;
+
+        prn.writef(" %s:", interface_to_str(interface));
+
+        while (size < LineSize) {
+            size += prn.writef(" %s%s%s", prefix, str, suffix);
+
+            str = list.nextof(str);
+            if (!str) {
+                break;
+            }
+        }
+
+        prn.writef("\n");
+    }
+}
+
+} // namespace
+
+bool print_supported(ProtocolMap& protocol_map, core::IArena& arena) {
+    core::Printer prn;
+    core::Array<Interface> interface_array(arena);
+    core::StringList list(arena);
+
+    if (!interface_array.resize(Iface_Max)) {
+        return false;
+    }
```

Comment:
This call is not quite correct: we resize() array here, but then we clear() it in get_supported_interfaces(), so it's size becomes zero again. It happens that it's capacity remains Iface_Max, but it's not obvious.

I suggest to remove resize() call here, and instead do the following in get_supported_interfaces() when we're adding interface to array:

```cpp
if (!interface_array.grow(interface_array.size() + 1)) {
    return false;
}
interface_array.push_back(protos_[y].iface);
```